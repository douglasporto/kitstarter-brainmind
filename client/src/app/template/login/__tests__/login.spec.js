import { shallowMount, createLocalVue } from '@vue/test-utils';
import flushPromises from 'flush-promises';
import bmLogin from '../main';
import VeeValidate from 'vee-validate';

let localVue = createLocalVue();
localVue.use(VeeValidate);

describe('Login.vue', () => {
  const build = () => {
    const wrapper = shallowMount(bmLogin, { localVue });
    return {
      wrapper
    };
  };

  it('Renders Login', () => {
    const { wrapper } = build();
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Expect data', () => {
    const { wrapper } = build();
    expect(wrapper.vm.email).toBe('diego3@rocketseat.com.br');
  });

  // it('Expect error E-mail', () => {
  //   const { wrapper } = build();
  //   wrapper.setData({
  //     email: 'douglas@'
  //   });
  //   console.log('doug', wrapper.find('.teste'));
  //   expect(wrapper.find('.teste')).toContain('Não cara, isso não é um E-mail');
  // });
  it('watches input value on model change', async () => {
    const { wrapper } = build();
    // const wrapper = shallowMount(bmLogin, { localVue });
    expect(wrapper.vm.errors.count()).toBe(0);
    // console.log(wrapper.vm.email);
    const input = wrapper.find('.form-group');
    input.trigger('input');
    console.log(input);
    wrapper.setData({ email: 'douglas' });
    wrapper.trigger('body');
    // console.log(wrapper.vm.errors);
    flushPromises();
    expect(wrapper.vm.errors.has('field')).toBe(true);
  });
});
