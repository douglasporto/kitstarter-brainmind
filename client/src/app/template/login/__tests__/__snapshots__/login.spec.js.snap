// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login.vue Renders Login 1`] = `
<section id="wrapper">
  <div class="login-register" style="background-image: url(undefinedimages/background/login-register.jpg);">
    <div class="login-box card">
      <div class="card-body">
        <div id="loginform" class="form-horizontal form-material">
          <h3 class="box-title m-b-20">Login</h3>
          <div class="form-group ">
            <div class="col-xs-12"><input name="email" type="text" placeholder="E-mail" class="form-control" aria-required="true" aria-invalid="false">
              <!---->
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
